The Mac grpc-tools build is broken with a CMake error

See [this job run](https://github.com/grpc/grpc-node/actions/runs/17653886027/job/50180805669?pr=3011) for an example failure.

Error message:

```
CMake Error at deps/protobuf/cmake/CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.
-- Configuring incomplete, errors occurred!

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

--- a/packages/grpc-tools/build_binaries.sh
+++ b/packages/grpc-tools/build_binaries.sh
@@ -36,7 +36,8 @@ build() {
   rm -f "$protobuf_base/CMakeCache.txt"
   rm -rf "$protobuf_base/CMakeFiles"
 
-  cmake -DCMAKE_TOOLCHAIN_FILE=linux.toolchain.cmake $cmake_flag . && cmake --build . --target clean && cmake --build . -- -j 12
+  # CMAKE_POLICY_VERSION_MINIMUM option is needed until the protobuf dependency is updated
+  cmake -DCMAKE_TOOLCHAIN_FILE=linux.toolchain.cmake -DCMAKE_POLICY_VERSION_MINIMUM=3.5 $cmake_flag . && cmake --build . --target clean && cmake --build . -- -j 12
 
   mkdir -p "$base/build/bin"
 

